### DISCUSSION TITLE
Issue with aideck wifi stream

### DISCUSSION THREAD

**Discussion by dhamidou** [bitcraze/discussions#1447] [Q&A]
I managed to get the aideck's [wifi](https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/) but when I connect to this wifi and run opencv-viewer.py, the code only displays :
Connecting to socket on 192.168.4.1:5000...
Socket connected
or at best opencv displays a black image or sometimes a few images but with a non-constant stream (one image in 10s, or 4 images in 10s then the stream freezes)
can you help me to understand this problem?
Thanks in advance

NB: Two days ago the stream was sending me an average of 7 images per second, but since I updated the aideck nothing works. And now, even if I change aideck or Crazyflie 2.X control board, I have this problem.

![Capture d’écran de 2024-07-04 10-39-55](https://github.com/bitcraze/discussions/assets/95414130/d26b8ba2-4445-4707-bf3b-90667e14b198)

**Comment by gemenerik** [comment]
To be clear; does nothing work anymore or does it sometimes work? 

Can you try reflashing default crazyflie firmware through the client? If you have the AI-deck attached this will also reflash the ESP32. And then please reflash the wifi streamer example to the gap8.

**Reply by dhamidou** [reply]
It works half the time. That is, sometimes I get 1 image in 10 seconds and sometimes I get no image at all. I try to reload crazyflie's default firmware via the client but nothing worked. The strange thing is that today everything works, whereas yesterday nothing worked. Between yesterday and today I just typed this command on my terminal: sudo apt install libxcb-xinerama0 but I already had it so I don't think there's any connection.

**Reply by gemenerik** [reply]
Happy to hear it works today. Let me know if it breaks again and if you changed anything

**Reply by dhamidou** [reply]
Hi Rik, after updating my PC (I'm on Ubuntu 20.04), I'm having the same problem again.

**Reply by gemenerik** [reply]
Can you check if the GAP8 initializes properly? The GAP8 LED should blink in this example.

Consider connecting the AI-deck to an existing network, rather than using it in AP mode.

**Reply by dhamidou** [reply]
The GAP8 LED blink at 1 Hz. 
I've just connected the aideck to my router but I'm having the same problem.

**Reply by gemenerik** [reply]
Is the ESP32 antenna still attached properly?

You tried reflashing all chips correct?

**Reply by dhamidou** [reply]
How can I check that the ESP32 is correctly attached? 
and I have reflashed all the chips using the following [steps](https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/)

**Reply by evoggy** [reply]
Are you connecting to the same WiFi network all the time or are you connecting to different networks (like office/lab/home)?

The antenna is the small metallic plate which is attached just at the edge of the PCB (as far back as possible).

**Reply by dhamidou** [reply]
yes, I'm always connected to the WiFi streaming example network.
Yes, the antenna is correctly attached.
What's strange is that I have the same problem with my 3 aidecks, 1 of which I've never used before.

**Reply by knmcguire** [reply]
Hi! Both Marcus and Rik are on holiday right now.

How does your console output look now that you are not able to connect anymore?

**Reply by dhamidou** [reply]
Hi, the problem is that I manage to connect to the aideck's wifi but I don't receive any images or sometimes I get 1 image in 20 sec.

![Capture d’écran de 2024-07-22 16-03-04](https://github.com/user-attachments/assets/c40ec749-1960-4e1d-8246-ef22cfc97c41)
![Capture d’écran de 2024-07-22 15-49-04](https://github.com/user-attachments/assets/d788e7de-8264-4f36-8320-56f0d30ad191)
![Capture d’écran de 2024-07-22 16-05-30](https://github.com/user-attachments/assets/2513c6d9-cfe7-4e30-819a-f017328c88b2)

**Reply by knmcguire** [reply]
I see that you are still using the wifi example. You can configure the crazyflie+aideck to connect to an access point instead, which might improve the stability, as the current setup can be also due to the wifi handling on your computer.

There are some instructions here that you can try out https://www.bitcraze.io/documentation/repository/aideck-gap8-examples/master/simple-examples/wifi-streamer/. You do need to be on a network that is not too restrictive in letting through access

**Reply by IVDC-IITI** [reply]
Even I am facing the same issue. It worked for a while and the it stopped receiving any data. It worked for merely few seconds.After which when i tried to reconnect and stream again it gave this error:
```
Connecting to socket on 192.168.4.1:5000...
Traceback (most recent call last):
  File "/home/nambiar/aideck-gap8-examples/examples/other/wifi-img-streamer/opencv-viewer.py", line 58, in <module>
    client_socket.connect((deck_ip, deck_port))
OSError: [Errno 113] No route to host
```
even though it was connected to the same wifi and port was working well.

When i killed the terminal when it stopped giving me data the program was stuck here for your reference:
```
^CTraceback (most recent call last):
  File "/home/nambiar/aideck-gap8-examples/examples/other/wifi-img-streamer/opencv-viewer.py", line 102, in <module>
    chunk = rx_bytes(length - 2)
  File "/home/nambiar/aideck-gap8-examples/examples/other/wifi-img-streamer/opencv-viewer.py", line 67, in rx_bytes
    data.extend(client_socket.recv(size-len(data)))
KeyboardInterrupt
```
After which then even if the socket is getting connected its showing no output..

![image](https://github.com/user-attachments/assets/b5614b2c-c01b-4d12-bb89-df9a5929e5e9)


I checked if it is an issue from the client side, like some firewall or any other, but no. 
Was facing the same issue as this highlighted by @dhamidou a day before, but suddenly worked for sometime and then the issue continued.
I tried flashing the software once again but in vein

This is the result of the `tcpdump` which i tried running alongside the `opencv-viewer.py`:
![image](https://github.com/user-attachments/assets/cb17a379-5d3f-4d0a-8810-63a087418d51)

**Reply by ToveRumar** [reply]
Hi @IVDC-IITI !

We did some changes to the streaming in our latest release. Please make sure to update to 2025.02, and during the update make sure that the ESP32 CPU is flashed with the latest binary. 

You can also test to connect it as a Wifi Station and see if this helps. Follow the guide "connect to wifi network" here; https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/#flash-wifi-example

I hope this helps your issue.

**Reply by IVDC-IITI** [reply]
I have followed the same tutorial @ToveRumar . The FW is flashed with 2025.02 version and have used the latest binary file from the release. It runs for about a minute and then the stream viewer just freezes, and then even after socket gets connected, no data is streamed

**Reply by gemenerik** [reply]
Thanks for confirming you updated to the latest release. Have you tried connecting to an existing WiFi network?

**Comment by gemenerik** [comment]
We've opened a dedicated issue to track the AI‑deck streaming freeze problem and consolidate efforts: https://github.com/bitcraze/aideck-gap8-examples/issues/150
